### DISCUSSION TITLE
snmp not pulling in switch status

### DISCUSSION THREAD

**Discussion by calvaradESU** [opendcim/openDCIM#1413] [Q&A] [answered]
I have a Cisco Catalyst 2960-X switch.  SNMP is configured and opendcim pulls in port names and notes OK, but it's not getting the port status.  Anybody have any ideas why?

**Comment by calvaradESU** [comment]
I've checked a few other switches and it's not pulling port status for any of them.  This is on a new install of opendcim v23.02.

**Reply by wilpig** [reply]
Send me an snmpwalk of the device

**Reply by calvaradESU** [reply]
snmpwalk of the status OID shows the statuses:

snmpwalk -v2c -c********* **.**.**.** 1.3.6.1.2.1.2.2.1.8


iso.3.6.1.2.1.2.2.1.8.1 = INTEGER: 1
iso.3.6.1.2.1.2.2.1.8.99 = INTEGER: 1
iso.3.6.1.2.1.2.2.1.8.5179 = INTEGER: 2
iso.3.6.1.2.1.2.2.1.8.5180 = INTEGER: 2
iso.3.6.1.2.1.2.2.1.8.5181 = INTEGER: 2
iso.3.6.1.2.1.2.2.1.8.10101 = INTEGER: 1
iso.3.6.1.2.1.2.2.1.8.10102 = INTEGER: 1
iso.3.6.1.2.1.2.2.1.8.10103 = INTEGER: 1
iso.3.6.1.2.1.2.2.1.8.10104 = INTEGER: 1
iso.3.6.1.2.1.2.2.1.8.10105 = INTEGER: 2
iso.3.6.1.2.1.2.2.1.8.10106 = INTEGER: 1
iso.3.6.1.2.1.2.2.1.8.10107 = INTEGER: 2
iso.3.6.1.2.1.2.2.1.8.10108 = INTEGER: 2
iso.3.6.1.2.1.2.2.1.8.10109 = INTEGER: 2
iso.3.6.1.2.1.2.2.1.8.10110 = INTEGER: 1
iso.3.6.1.2.1.2.2.1.8.10111 = INTEGER: 1
iso.3.6.1.2.1.2.2.1.8.10112 = INTEGER: 1
iso.3.6.1.2.1.2.2.1.8.10113 = INTEGER: 1
iso.3.6.1.2.1.2.2.1.8.10114 = INTEGER: 1
iso.3.6.1.2.1.2.2.1.8.10115 = INTEGER: 1
iso.3.6.1.2.1.2.2.1.8.10116 = INTEGER: 1
iso.3.6.1.2.1.2.2.1.8.10117 = INTEGER: 1
iso.3.6.1.2.1.2.2.1.8.10118 = INTEGER: 2
iso.3.6.1.2.1.2.2.1.8.10119 = INTEGER: 2
iso.3.6.1.2.1.2.2.1.8.10120 = INTEGER: 2
iso.3.6.1.2.1.2.2.1.8.10121 = INTEGER: 2
iso.3.6.1.2.1.2.2.1.8.10122 = INTEGER: 2
iso.3.6.1.2.1.2.2.1.8.10123 = INTEGER: 2
iso.3.6.1.2.1.2.2.1.8.10124 = INTEGER: 2
iso.3.6.1.2.1.2.2.1.8.10125 = INTEGER: 2
iso.3.6.1.2.1.2.2.1.8.10126 = INTEGER: 2
iso.3.6.1.2.1.2.2.1.8.10127 = INTEGER: 2
iso.3.6.1.2.1.2.2.1.8.10128 = INTEGER: 2
iso.3.6.1.2.1.2.2.1.8.10129 = INTEGER: 2
iso.3.6.1.2.1.2.2.1.8.10130 = INTEGER: 2
iso.3.6.1.2.1.2.2.1.8.10131 = INTEGER: 2
iso.3.6.1.2.1.2.2.1.8.10132 = INTEGER: 2
iso.3.6.1.2.1.2.2.1.8.10133 = INTEGER: 2
iso.3.6.1.2.1.2.2.1.8.10134 = INTEGER: 2
iso.3.6.1.2.1.2.2.1.8.10135 = INTEGER: 2
iso.3.6.1.2.1.2.2.1.8.10136 = INTEGER: 2
iso.3.6.1.2.1.2.2.1.8.10137 = INTEGER: 2
iso.3.6.1.2.1.2.2.1.8.10138 = INTEGER: 2
iso.3.6.1.2.1.2.2.1.8.10139 = INTEGER: 2
iso.3.6.1.2.1.2.2.1.8.10140 = INTEGER: 2
iso.3.6.1.2.1.2.2.1.8.10141 = INTEGER: 2
iso.3.6.1.2.1.2.2.1.8.10142 = INTEGER: 2
iso.3.6.1.2.1.2.2.1.8.10143 = INTEGER: 2
iso.3.6.1.2.1.2.2.1.8.10144 = INTEGER: 2
iso.3.6.1.2.1.2.2.1.8.10145 = INTEGER: 2
iso.3.6.1.2.1.2.2.1.8.10146 = INTEGER: 1
iso.3.6.1.2.1.2.2.1.8.10147 = INTEGER: 1
iso.3.6.1.2.1.2.2.1.8.10148 = INTEGER: 2
iso.3.6.1.2.1.2.2.1.8.10149 = INTEGER: 2
iso.3.6.1.2.1.2.2.1.8.10150 = INTEGER: 2
iso.3.6.1.2.1.2.2.1.8.10201 = INTEGER: 1
iso.3.6.1.2.1.2.2.1.8.10202 = INTEGER: 2
iso.3.6.1.2.1.2.2.1.8.14001 = INTEGER: 1
iso.3.6.1.2.1.2.2.1.8.14002 = INTEGER: 2

**Reply by calvaradESU** [reply]
Any thoughts?  Is there a switch accessible to the demo site for testing this?

**Reply by samilliken** [reply]
No, there is not a demo switch accessible from the demo website.

**Reply by wilpig** [reply]
While not the demo site I have one connected to the dev site at https://dev.opendcim.org/devices.php?DeviceID=4788 .  The snmp walk you have looks the same as what I am getting from another switch.

```
$ snmpwalk -v2c -On -c pig 10.0.0.5 .1.3.6.1.2.1.2.2.1.8
.1.3.6.1.2.1.2.2.1.8.1 = INTEGER: up(1)
.1.3.6.1.2.1.2.2.1.8.2 = INTEGER: up(1)
.1.3.6.1.2.1.2.2.1.8.3 = INTEGER: down(2)
.1.3.6.1.2.1.2.2.1.8.4 = INTEGER: down(2)
.1.3.6.1.2.1.2.2.1.8.5 = INTEGER: down(2)
.1.3.6.1.2.1.2.2.1.8.6 = INTEGER: down(2)
.1.3.6.1.2.1.2.2.1.8.7 = INTEGER: up(1)
.1.3.6.1.2.1.2.2.1.8.8 = INTEGER: down(2)
.1.3.6.1.2.1.2.2.1.8.9 = INTEGER: up(1)
.1.3.6.1.2.1.2.2.1.8.10 = INTEGER: up(1)
.1.3.6.1.2.1.2.2.1.8.11 = INTEGER: up(1)
.1.3.6.1.2.1.2.2.1.8.12 = INTEGER: up(1)
.1.3.6.1.2.1.2.2.1.8.13 = INTEGER: up(1)
.1.3.6.1.2.1.2.2.1.8.14 = INTEGER: up(1)
.1.3.6.1.2.1.2.2.1.8.15 = INTEGER: down(2)
.1.3.6.1.2.1.2.2.1.8.16 = INTEGER: down(2)
.1.3.6.1.2.1.2.2.1.8.17 = INTEGER: down(2)
.1.3.6.1.2.1.2.2.1.8.18 = INTEGER: down(2)
.1.3.6.1.2.1.2.2.1.8.19 = INTEGER: up(1)
.1.3.6.1.2.1.2.2.1.8.20 = INTEGER: down(2)
.1.3.6.1.2.1.2.2.1.8.21 = INTEGER: up(1)
.1.3.6.1.2.1.2.2.1.8.22 = INTEGER: down(2)
.1.3.6.1.2.1.2.2.1.8.23 = INTEGER: up(1)
.1.3.6.1.2.1.2.2.1.8.24 = INTEGER: up(1)
.1.3.6.1.2.1.2.2.1.8.25 = INTEGER: up(1)
.1.3.6.1.2.1.2.2.1.8.26 = INTEGER: up(1)
.1.3.6.1.2.1.2.2.1.8.65 = INTEGER: up(1)
.1.3.6.1.2.1.2.2.1.8.66 = INTEGER: up(1)
.1.3.6.1.2.1.2.2.1.8.67 = INTEGER: up(1)
.1.3.6.1.2.1.2.2.1.8.68 = INTEGER: down(2)
.1.3.6.1.2.1.2.2.1.8.69 = INTEGER: down(2)
.1.3.6.1.2.1.2.2.1.8.70 = INTEGER: down(2)
.1.3.6.1.2.1.2.2.1.8.71 = INTEGER: down(2)
.1.3.6.1.2.1.2.2.1.8.72 = INTEGER: down(2)
.1.3.6.1.2.1.2.2.1.8.73 = INTEGER: down(2)
.1.3.6.1.2.1.2.2.1.8.74 = INTEGER: down(2)
.1.3.6.1.2.1.2.2.1.8.75 = INTEGER: down(2)
.1.3.6.1.2.1.2.2.1.8.76 = INTEGER: down(2)
.1.3.6.1.2.1.2.2.1.8.77 = INTEGER: down(2)
.1.3.6.1.2.1.2.2.1.8.78 = INTEGER: down(2)
.1.3.6.1.2.1.2.2.1.8.79 = INTEGER: down(2)
.1.3.6.1.2.1.2.2.1.8.80 = INTEGER: down(2)
.1.3.6.1.2.1.2.2.1.8.81 = INTEGER: down(2)
```

The difference in my output and yours is the port value.  Did you set the first port id on the device page?  Debugging this comes into several stages but the most simple would be:
1. goto the device page that I have linked above.  
2. Open the debugger
3. click on refresh port status
4. look at the output of the post job
![Screenshot 2023-08-29 at 11 22 30 AM](https://github.com/opendcim/openDCIM/assets/1653882/1e1799cf-d3fb-4c79-b5be-dd048117b73a)


All the data comes back through this call https://github.com/opendcim/openDCIM/blob/master/classes/SwitchInfo.class.php#L146C3-L146C3

**Comment by calvaradESU** [comment]
I get this error on my site:

devices.php?DeviceID=152:420  Uncaught TypeError: $.post(...).done(...).error is not a function
    at getfirstport (devices.php?DeviceID=152:420:6)
    at HTMLButtonElement.<anonymous> (devices.php?DeviceID=152:385:3)
    at HTMLButtonElement.dispatch (jquery.min.js:2:43064)
    at v.handle (jquery.min.js:2:41048)

but not on your site.

As you can see, the first port on my device is set to 6.

**Reply by wilpig** [reply]
https://github.com/opendcim/openDCIM/commit/5a783a254098e79e736575a5c7389de419218be3

Make this change from .error to .fail and see if you're getting a different result in the choices for the first port.

**Comment by calvaradESU** [comment]
After changing those lines, the switch ports look the same to me, but I don’t see that error any more.

***@***.***

**Comment by calvaradESU** [comment]
The port names and notes come in correctly, beginning with the correct port number.  It’s just the port status that isn’t working.

**Comment by timlegge** [comment]
Make sure you have the switch set to poll.  Depending on your value of the
polling under reporting, it will not update the status even though
everything else works.

At least that's the way it was with prior versions.

**Comment by calvaradESU** [comment]
Timothy,

Is this where you mean?
***@***.***

**Comment by timlegge** [comment]
I couldn't see what you posted. Attach a picture to the issue on GitHub instead of replying an email. Regardless, it's under settings reporting I believe. One way for the setting polls every device that is capable of being polled. The other way requires you to have a tag set on the device equal to poll I believe

**Reply by calvaradESU** [reply]
In my Configuration page, under Reporting, Capacity Reporting, Switches: is set to OptIn.  Is that what you meant?

I don't see any checkbox or option on the device or the template to turn polling on or off.

**Reply by timlegge** [reply]
Yes.  Go to that device and enter a tag named "Poll"

**Comment by calvaradESU** [comment]
On the device, if I enter a tag named Poll and click Update, the Poll does not stay.  My only options in the Tags drop list is “Report” and “NoReport”.

**Reply by timlegge** [reply, marked as answer]
It's possible that report might work. I forget exactly. I think there were two options. To enter a new tag called poll type Poll and press enter. Then when you save it it should stay

**Reply by calvaradESU** [reply]
THANK YOU Timothy!!!  That Worked!!
